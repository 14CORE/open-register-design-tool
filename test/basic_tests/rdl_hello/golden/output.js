/*   Ordt 170413.03 autogenerated file 
 *   Input: ./rdl_hello/test.rdl
 *   Parms: ./rdl_hello/test.parms
 *   Date: Thu Apr 13 23:28:51 EDT 2017
 */

register_set foo "foo registers" {

  address = 0x0;
  register_width = 32;
  description = "{
    this is a bunch of foo regs bla bla bla
  }";
  register_set bar "bar registers" {
    address = 0x0;
    description = "{
      this is a bunch of bar regs bla bla bla
    }";
    
    register a_reg "a_reg register" {
      address = 0x0;
      repeat = 2;
      nop[16];
      integer fld2[1] "field 2" param {
        access_mode = READ_WRITE;
        reset = unknown;
      };
      nop[5];
      integer fld1[10] "field 1" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register_set_size = 0x8;
  };
  
  register_set_size = 0x8;
};

