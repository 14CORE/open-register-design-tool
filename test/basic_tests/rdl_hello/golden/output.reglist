//   Ordt 160614.01 autogenerated file 
//   Input: ./rdl_hello/test.rdl
//   Parms: ./rdl_hello/test.parms
//   Date: Tue Jun 14 16:31:50 EDT 2016
//

0x0              size: 32   fields: 2    bits: 11        bar.a_reg_0
0x4              size: 32   fields: 2    bits: 11        bar.a_reg_1
