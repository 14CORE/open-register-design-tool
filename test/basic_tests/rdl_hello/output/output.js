/*   Ordt 160614.01 autogenerated file 
 *   Input: ./rdl_hello/test.rdl
 *   Parms: ./rdl_hello/test.parms
 *   Date: Wed Jun 15 10:26:39 EDT 2016
 */

register_set foo "Registers for foo" {

  address = 0x0;
  register_width = 32;
  register_set bar "bar registers" {
    address = 0x0;
    description = "{
      this is a bunch of bar regs bla bla bla
    }";
    
    register a_reg "a_reg register" {
      address = 0x0;
      repeat = 2;
      nop[16];
      integer fld2[1] "field 2" param {
        access_mode = READ_WRITE;
        reset = unknown;
      };
      nop[5];
      integer fld1[10] "field 1" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register_set_size = 0x8;
  };
  
  register_set_size = 0x8;
};

