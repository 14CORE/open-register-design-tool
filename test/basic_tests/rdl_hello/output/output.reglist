//   Ordt 160614.01 autogenerated file 
//   Input: ./rdl_hello/test.rdl
//   Parms: ./rdl_hello/test.parms
//   Date: Wed Jun 15 10:26:39 EDT 2016
//

0x0              size: 32   fields: 2    bits: 11        bar.a_reg_0
0x4              size: 32   fields: 2    bits: 11        bar.a_reg_1
