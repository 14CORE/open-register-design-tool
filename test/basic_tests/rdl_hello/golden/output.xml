<?xml version="1.0" encoding="UTF-8"?>
<!--   Ordt 160614.01 autogenerated file 
       Input: ./rdl_hello/test.rdl
       Parms: ./rdl_hello/test.parms
       Date: Tue Jun 14 16:31:50 EDT 2016
 -->

<map version="160614.01">
  <id>foo</id>
  <baseaddr>0</baseaddr>
  <shorttext>foo registers</shorttext>
  <regset>
    <id>bar</id>
    <shorttext>bar registers</shorttext>
    <baseaddr>0x0</baseaddr>
    <longtext><![CDATA[this is a bunch of bar regs bla bla bla]]></longtext>
    <reg>
      <id>a_reg</id>
      <parentpath>bar</parentpath>
      <shorttext>a_reg register</shorttext>
      <baseaddr>0x0</baseaddr>
      <width>32</width>
      <reps>2</reps>
      <stride>0x4</stride>
      <field>
        <id>fld1</id>
        <shorttext>field 1</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>0</lowidx>
        <width>10</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld2</id>
        <shorttext>field 2</shorttext>
        <access>RW</access>
        <lowidx>15</lowidx>
        <width>1</width>
      </field>
      <access>RW</access>
    </reg>
    <highaddr>0x7</highaddr>
  </regset>
</map>
